### PR TITLE
Improve DPI aware builds with .vc and .vcproj files

### PR DESCRIPTION
**Pull request by MaartenBent**
I tried to address the issues in https://trac.wxwidgets.org/ticket/18665 and https://trac.wxwidgets.org/ticket/18640.

For 18665, I disabled embedding the DPI manifest in the `.vcproj` files. In the `.vc` files, it is enabled for VS2015, VS2017 and VS2019 (based on the `VISUALSTUDIOVERSION` env. var. set by the developer command prompt).
The downside is that there are no `vcxproj` files (except the minimal sample), so when the `.vcproj` files are opened and upgraded by a newer VS, it will not have the DPI aware manifest.

For 18640, I added an option to automatically detect the `TARGET__CPU` based on the `PLATFORM` env. var. If it cannot be determined, a build error is generated informing the user that `CPU` should be specified.

Als enable DPI awareness in demos with CMake build.

### CODE REVIEW

File: build/bakefiles/common.bkl [hunk 1]
```diff
@@ -344,19 +344,39 @@
...
+            <if cond="USE_DPI_AWARE_MANIFEST=='1' and VS_VER=='16.0' and TARGET_CPU=='x64'">/MANIFEST:EMBED /MANIFESTINPUT:$(TOP_SRCDIR)include/wx/msw/amd64_dpi_aware.manifest</if>
+            <if cond="USE_DPI_AWARE_MANIFEST=='1' and VS_VER=='14.0' and TARGET_CPU=='X64'">/MANIFEST:EMBED /MANIFESTINPUT:$(TOP_SRCDIR)include/wx/msw/amd64_dpi_aware.manifest</if>
+            <if cond="USE_DPI_AWARE_MANIFEST=='1' and VS_VER=='15.0' and TARGET_CPU=='X64'">/MANIFEST:EMBED /MANIFESTINPUT:$(TOP_SRCDIR)include/wx/msw/amd64_dpi_aware.manifest</if>
+            <if cond="USE_DPI_AWARE_MANIFEST=='1' and VS_VER=='16.0' and TARGET_CPU=='X64'">/MANIFEST:EMBED /MANIFESTINPUT:$(TOP_SRCDIR)include/wx/msw/amd64_dpi_aware.manifest</if>
             <!-- 32-bit per-monitor dpi -->
-            <if cond="USE_DPI_AWARE_MANIFEST=='2' and TARGET_CPU==''">/MANIFEST:EMBED /MANIFESTINPUT:$(TOP_SRCDIR)include/wx/msw/wx_dpi_aware_pmv2.manifest</if>
+            <if cond="USE_DPI_AWARE_MANIFEST=='2' and VS_VER=='14.0' and TARGET_CPU==''">/MANIFEST:EMBED /MANIFESTINPUT:$(TOP_SRCDIR)include/wx/msw/wx_dpi_aware.manifest</if>
```
Comment (MaartenBent) [inline build/bakefiles/common.bkl]: Copy paste error. Should be `wx_dpi_aware_pmv2.manifest` and `amd64_dpi_aware_pmv2.manifest`, will fix later.

File: build/bakefiles/config.bkl [hunk 2]
```diff
@@ -132,6 +132,14 @@ Accepted values: IA64, X64, ARM64
 (AMD64 accepted as synonym for X64 but should not be used any more).
             </description>
         </option>
+        <option name="VS_VER">
+            <!-- defined by VS command prompt, can (at least) be 14.0, 15.0, 16.0, ... -->
```
Comment (vadz) [inline build/bakefiles/config.bkl]: I initially misunderstood this as meaning that `VS_VER` itself was supposed to be defined in the environment. Now I see that it's not the case and that its value just comes from `VisualStudioVersion` environment variable, but maybe it would be better to reword this to avoid the confusion?

Also, while checking this I noticed that MSVS 2017 and 2019 define `VSCMD_ARG_HOST_ARCH` and `VSCMD_ARG_TGT_ARCH` env vars too (and `VSCMD_VER`, but this is less important). Unfortunately MSVS 2015 doesn't have those.

**Review by vadz — COMMENTED** [review]
Thanks a lot for taking care of this!

I don't have any real comments, but I'd just like to mention that we probably should add `.vcxproj` files for all the samples. It's going to be painful to maintain them, of course, but it's a bit ridiculuous to have MSVS 2003 projects and not MSVS 201x in 2020, so I think we'll still have them in 3.2 release. I'll take `samples/minimal/minimal.vcxproj` if I have to create them (semi)manually, so the only thing I could ask for would be to make sure this project does the right thing for all the different MSVS 201x versions. Or is this already the case?

Thanks again and I'll merge it as soon as you fix the copy-paste typo you noticed.

**Comment by MaartenBent** [comment]
The `.vcxproj` files use the `AdditionalManifestFiles` option, not the problematic `/MANIFEST:EMBED /MANIFESTINPUT:file.manifest`, so it should be fine.
But it would be great if someone could test it with older (2010, 2012, 2015) Visual Studio versions.

If it doesn't work, it can simply be disabled with a condition like:
```
<AdditionalManifestFiles Condition="'$(wxUseDPIManifest)'=='1'">./../../include/wx/msw/wx_dpi_aware_pmv2.manifest %(AdditionalManifestFiles)</AdditionalManifestFiles>
```
and in `wx_setup.props`:
```
<PropertyGroup Label="UserMacros">
  <wxUseDPIManifest Condition="'$(PlatformToolset)'!='v100' And '$(PlatformToolset)'!='v100_xp' And '$(PlatformToolset)'!='v110' And '$(PlatformToolset)'!='v110_xp' And '$(PlatformToolset)'!='v120' And '$(PlatformToolset)'!='v120_xp'">1</wxUseDPIManifest>
</PropertyGroup>
 ```

File: build/bakefiles/config.bkl (hunk 2 above)
Comment (MaartenBent) [reply build/bakefiles/config.bkl]: I renamed the options.

I saw those env vars too, and indeed 2015 does not have them.
